source/common/upstream/health_checker_base_impl.cc

```diff
@@ -132,6 +143,7 @@ void HealthCheckerImplBase::refreshHealthyStat() {
   // Each hot restarted process health checks independently. To make the stats easier to read,
   // we assume that both processes will converge and the last one that writes wins for the host.
   stats_.healthy_.set(local_process_healthy_);
+  stats_.degraded_.set(local_process_degraded_);
```

Comment:
cc @jmarantz I would take a look at this PR. This is a very succinct example of the potential differences in gauges if we get rid of shard memory on hot restart.